version: '3.3'

networks:
  error-network:
      driver: bridge

volumes:
  esdata:
    driver: local

services:
  ruby:
    build:
      context: .
      dockerfile: docker/ruby/Dockerfile
    volumes:
      - ./src:/app
      - ./docker/ruby/start.sh:/start.sh
    networks:
      - error-network
    ports:
      - "3000:3000"
    entrypoint:
      - /start.sh
  mysql:
    image: mysql:5.7
    ports:
      - "3306:3306"
    volumes:
      - ./src/database/import:/docker-entrypoint-initdb.d
    environment:
      - MYSQL_ROOT_PASSWORD=root
      - MYSQL_DATABASE=layers-app_development
      - MYSQL_USER=user
      - MYSQL_PASSWORD=root
    networks:
      - error-network
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      interval: 1s
      timeout: 1s
      retries: 300
  redis:
    image: redis:3.2
    networks:
      - error-network
    ports:
      - "6379:6379"
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 1s
      timeout: 1s
      retries: 300